### ISSUE TITLE
Unbound type variable false positive with six.with_metaclass

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#14475]
This code generates a false positive:
```py
import six
import abc
from typing import TypeVar, Generic

T = TypeVar("T")

class C(six.with_metaclass(abc.ABCMeta, Generic[T])):
    pass

class D(six.with_metaclass(abc.ABCMeta, C[T])):  # Type variable "t.T" is unbound
    pass
```
This is a regression from 0.991 introduced here:
* #14159

@ilevkivskyi Do you have time to look at this? This is a blocker for the 1.0 release.

**Comment by ilevkivskyi** [comment]
Here is the fix.
